desktop/src/main/java/bisq/desktop/primary/main/content/trade/bisqEasy/chat/BisqEasyChatController.java

```diff
@@ -143,6 +163,45 @@ protected void handleChannelChange(Channel<? extends ChatMessage> channel) {
         });
     }
 
+    private void updateLastOfferMessage() {
+        PrivateTradeChatMessage newLastOfferMessage = tradeChannelSelectionService.getSelectedChannel().get().getChatMessages().stream()
+                .filter(chatMessage -> chatMessage instanceof PrivateTradeChatMessage)
+                .map(chatMessage -> (PrivateTradeChatMessage) chatMessage)
+                .filter(message -> message.getChannelName().equals(tradeChannelSelectionService.getSelectedChannel().get().getChannelName()))
+                .filter(PrivateTradeChatMessage::hasTradeChatOffer)
+                .max(Comparator.comparing(PrivateTradeChatMessage::getDate))
+                .orElse(null);
+
+        if (newLastOfferMessage == lastOfferMessage) return;
+
+        lastOfferMessage = newLastOfferMessage;
+        updateTradeHelper();
+    }
+
+    private void updateTradeHelper() {
+        UIThread.runOnNextRenderFrame(() -> {
+            // Don't show complete trade button for BTC sellers or chat with no offers
+            if (lastOfferMessage == null) {
+                model.getCompleteTradeDisabled().set(true);
+                return;
+            }
+
+            checkArgument(lastOfferMessage.getTradeChatOffer().isPresent());
+            var offer = lastOfferMessage.getTradeChatOffer().get();
+
+            boolean isBtcSeller = isMyMessage(lastOfferMessage) == offer.getDirection().isBuy();
+            boolean noWallet = walletService.isEmpty();
+            boolean isDisabled = isBtcSeller || noWallet;
+            model.getCompleteTradeDisabled().set(isDisabled);
+
+            String tooltip = isBtcSeller ? "Coming soon" :
+                    noWallet ?
+                            Res.get("wallet.unavailable") :
+                            Res.get("bisqEasy.completeOffer.tooltip");
+            view.updateCompleteTradeTooltip(isDisabled, tooltip);
```

Comment:
In the MCV pattern we use we do not call the view from the controller, but change state in the model and the view is listening to those property changes.

Reply:
In this case you have already the disable property, just missing the tooltip property and a binding in the view.